coalib/results/TextPosition.py

```diff
@@ -31,3 +31,30 @@ def line(self):
     @property
     def column(self):
         return self._column
+
+    @classmethod
+    def calc_line_col(cls, text, pos_to_find):
+        """
+        Calculate line number and column in the text, from position.
+        Uses \\n as the newline character. Lines and columns start from 1.
+
+        :param text:        A string with all the contents of file.
+        :param pos_to_find: Position of character to be found in the
+                            (line, column) form.
+
+        :return:            a tuple of the form (line, column).
+        """
+        line = 1
+        pos = -1
+        pos_new_line = text.find('\n')
+        while True:
```

Comment:
What I meant was let's not take `text` as a big string. Rather let's take it as a `tuple of lines` (Like how `file` in Bear is)
So, this would need to be like:
```
    @classmethod
    def calc_line_col(cls, text, pos_to_find):
        """
        Calculate line number and column in the text, from position.
        Uses \\n as the newline character. Lines and columns start from 1.

        :param text:        A tuple of lines.
        :param pos_to_find: Position of character to be found in the
                            (line, column) form.
        :return:            a tuple of the form (line, column).
        """
        line = col = 0
        for textline in text:
            new_position = position - len(line)
            if new_position < 0:
                break
            line += 1
            position = new_position
        col = position
        return line, col
```
PS: I have not checked for corner cases.

PPS: Now, in `calc_position` you don't need to do a `splitlines()` as both these functions use a list of lines and not a huge string chunk.